autoload/go/test.vim

```diff
@@ -122,13 +122,53 @@ function! go#test#Func(bang, ...) abort
   call call('go#test#Test', args)
 endfunction
 
+" Testcase runs a single subtest of a function for a subtest name on the
+" current cursor line.
+" Arguments are passed to the `go test` command.
+function! go#test#Case(bang, ...) abort
+  let l:testParent = go#util#TestName()
+  if l:testParent == ""
+    call go#util#EchoWarning("[test] no test found immediate to cursor")
+    return
+  endif
+
+  " Check for a struct initialization on the line, and take its value
+  " e.g. `name: \"test name\",` would extract 'testName'
+  let l:subTestName = go#test#subtestUnderCursor(getline("."))
+  if l:subTestName == ""
+    call go#util#EchoError('[test] no name for test case found on cursor line')
+    return
+  endif
+
+  " the pattern after l:testParent will allow testcases that are nested
+  " arbitrarily deep to be selected to run
+  let l:testParentPattern = l:testParent . "/\([^/]\+/\)*"
+
+  let l:args = [a:bang, 0, "-run", l:testParentPattern . "/" . l:subTestName . "$"]
+  if a:0
+    call extend(l:args, a:000)
+  else
+    let l:timeout = go#config#TestTimeout()
+    call add(l:args, printf("-timeout=%s", l:timeout))
+  endif
+
+  call call('go#test#Test', l:args)
+endfunction
+
 function! s:test_job(cmd, args) abort
   " autowrite is not enabled for jobs
   call go#cmd#autowrite()
 
   call go#job#Spawn(a:cmd, a:args)
 endfunction
 
+function! go#test#subtestUnderCursor(cursorLine) abort
+  " e.g. `name: \"test name\",` would extract 'test name'
+  " e.g. `name: \"test_name\"` would extract 'test_name'
+  let l:structInitPattern =  '\:\s*\"\zs\(\w\|\s\)*\ze\"'
```

Comment:
This is still too narrow. Tests names can easily be outside the set you've defined here. Consider:
* `name: "foo-bar",`
* `name: "México",`
* `name: "⌘",`
* `name: "thing 1, thing 2, and thing 3",`
*  `name: "condition a; condition b",`
* `name: "1/2",`

I don't mean for that list to be exhaustive; it's only illustrative of some examples in the broader scope that this regular expression needs to be constructed to handle.

I also wonder if this needs to handle raw strings in addition to the usual double quoted strings..

Reply:
There's no need to try to leverage LSP for this.

Try `:\s*\%("\([^"]\+\)"\)` and if that doesn't match anything then try that same, but with backticks instead of quotes.

> calls that are called before the calls for the test cases

What do you mean?

> strings that are transformed from the declaration in the test case to something different when running the test; `t.Run(fmt.Sprintf("Test_%s", testCase.name), func( tt *testing.T) { ...`

Yeah, that's a concern. And is one of the kinds of edges that historically kept this feature from being implemented.

Reply:
> What do you mean?
It's mostly a moot point if you don't want LSP usage here, and likely an extreme corner case. But regarding 
> if > 1 string symbols exist find the next invocation of testing.Run() and check for the value passed to that and search symbols on the cursor line for a string symbol directly after that field symbol.
and identifying the `t.Run(...` that is being looped for the test cases to discover the field used for the name. 
```
func Awesome_Test(t *testing.T) {
    testCases := []struct{
        name string
        want string
     }{
         // unless we key specifically for the field or string under the cursor this is ambiguous for which
         // field should be the test name
         {name: "myTest", want: "myValue"},
         {name: "yourTest", want: "yourValue"},
     }

     // This test doesn't use the test cases but would still show a non string symbol for the value. So checking 
     // for that symbol in the fields of the test cases will at best return nothing, and at worst return the wrong value
     // to test with.
     // There's more complexity if several sets of test cases are also declared in a single test function as well. 
     subTestName := "foo"
     t.Run(subTestName, func(tt *testing.T){
       // test logic
    }

    // this would be the actual testing.Run() call we'd want to check to identify the field used for the test name in
    // in the testing struct. 
    for _, tc := range testCases {
        t.Run(tc.name, func(tt *testing.T){
            // test logic
        }
    }
}
```
But like I noted above, if you don't want LSP usage for this feature that's all moot. 


Otherwise IntelliJ(GoLand) has some [restrictions noted](https://www.jetbrains.com/go/guide/tips/run-test-in-table-test/) for using this feature that would get around issues about case manipulation between the declaration of the test cases and running them. But I'm not sure your appetite for restricting use cases for a feature. 

I think the string formatting problem can be worked around with an inclusive enough pattern in the `go test -run ...` call. But I'm going to tear this PR down and rework a couple of things to see if I can get this to be more flexible.